components/root-actions/MoveAuthoredContributions.js

```diff
@@ -0,0 +1,315 @@
+import ConfirmationModal from '../ConfirmationModal';
+import Container from '../Container';
+import { Box, Flex } from '../Grid';
+import LinkCollective from '../LinkCollective';
+import MessageBox from '../MessageBox';
+import MessageBoxGraphqlError from '../MessageBoxGraphqlError';
+import StyledButton from '../StyledButton';
+import StyledCheckbox from '../StyledCheckbox';
+import StyledInputField from '../StyledInputField';
+import StyledLink from '../StyledLink';
+import StyledSelect from '../StyledSelect';
+import StyledTag from '../StyledTag';
+import { Label, P, Span } from '../Text';
+import { TOAST_TYPE, useToasts } from '../ToastProvider';
+
+const moveOrdersFieldsFragment = gqlV2/* GraphQL */ `
+  fragment MoveOrdersFields on Order {
+    id
+    legacyId
+    description
+    createdAt
+    amount {
+      valueInCents
+      currency
+    }
+    fromAccount {
+      id
+      name
+      slug
+      isIncognito
+      imageUrl(height: 48)
+    }
+    toAccount {
+      id
+      slug
+      name
+    }
+  }
+`;
+
+const ordersQuery = gqlV2/* GraphQL */ `
+  query AuthoredOrdersRootQuery($account: AccountReferenceInput!) {
+    orders(account: $account, filter: OUTGOING, limit: 100, includeIncognito: true) {
+      nodes {
+        ...MoveOrdersFields
+      }
+    }
+  }
+  ${moveOrdersFieldsFragment}
+`;
+
+const moveOrdersMutation = gqlV2/* GraphQL */ `
+  mutation MoveOrdersMutation(
+    $orders: [OrderReferenceInput!]!
+    $fromAccount: AccountReferenceInput!
+    $makeIncognito: Boolean
+  ) {
+    moveOrders(orders: $orders, fromAccount: $fromAccount, makeIncognito: $makeIncognito) {
+      ...MoveOrdersFields
+    }
+  }
+  ${moveOrdersFieldsFragment}
+`;
+
+const getOrdersOptionsFromData = (intl, data) => {
+  if (!data?.orders) {
+    return [];
+  }
+
+  return data.orders.nodes.map(order => {
+    const date = intl.formatDate(order.createdAt);
+    const amount = formatCurrency(order.amount.valueInCents, order.amount.currency, { locale: intl.locale });
+    return {
+      value: order,
+      label: `${date} - ${amount} contribution to @${order.toAccount.slug} (#${order.legacyId})`,
+    };
+  });
+};
+
+const getCallToAction = (selectedOrdersOptions, newFromAccount) => {
+  if (newFromAccount?.isIncognito) {
+    return `Mark ${selectedOrdersOptions.length} contributions as incognito`;
+  } else {
+    const base = `Move ${selectedOrdersOptions.length} contributions`;
+    return !newFromAccount ? base : `${base} to @${newFromAccount.slug}`;
+  }
+};
+
+const getToAccountCustomOptions = fromAccount => {
+  if (!fromAccount) {
+    return [];
+  }
+
+  const incognitoLabel = `@${fromAccount.slug}'s incognito profile`;
+  return [
+    {
+      [FLAG_COLLECTIVE_PICKER_COLLECTIVE]: true,
+      value: fromAccount,
+    },
+    {
+      [FLAG_COLLECTIVE_PICKER_COLLECTIVE]: true,
+      label: incognitoLabel,
+      value: { name: incognitoLabel, useIncognitoProfile: true, isIncognito: true },
+    },
+  ];
+};
```

Comment:
I think here we show the incognito profile for all users in the `Move to` dropdown. However note that only individual users can have incognito profiles. So trying to choose the "incognito" profile of a org will result in the following error always. 

https://github.com/opencollective/opencollective-api/blob/fe9403c87718298812aaa9eb6dc1f917ddc32b94/server/models/Collective.js#L926-L928. 

So I think we need to have a condition in the front-end only to show this incognito profile option when author of the contribution is a user (not a org).

Reply:
Good catch!